django/forms/boundfield.py

```diff
@@ -289,20 +289,26 @@ def build_widget_attrs(self, attrs, widget=None):
             attrs["disabled"] = True
         if not widget.is_hidden and self.errors:
             attrs["aria-invalid"] = "true"
-        # If a custom aria-describedby attribute is given (either via the attrs
-        # argument or widget.attrs) and help_text is used, the custom
-        # aria-described by is preserved so user can set the desired order.
-        if (
-            not attrs.get("aria-describedby")
-            and not widget.attrs.get("aria-describedby")
-            and self.field.help_text
-            and not self.use_fieldset
-            and self.auto_id
-            and not self.is_hidden
-        ):
-            attrs["aria-describedby"] = f"{self.auto_id}_helptext"
+        # Preserve aria-describedby provided by the attrs argument so user
+        # can set the desired order.
+        if not attrs.get("aria-describedby") and not self.use_fieldset:
+            if aria_describedby := self.aria_describedby:
+                attrs["aria-describedby"] = aria_describedby
         return attrs
 
+    @property
```

Comment:
Maybe we use `@cached_property`  so that `{% if field.aria_describedby %} aria-describedby="{{ field.aria_describedby }}"{% endif %}` builds the string once :thinking:

Reply:
Thinking about this today. I'm not so sure -- see the example below the final output doesn't have the error reference. 

I appreciate this is a bit tenuous. 

``` python
In [6]: class ContactForm(forms.Form):
   ...:     name = forms.CharField(help_text="Help Text!")
   ...:

In [7]: f = ContactForm({'name':"David"})

In [8]: print(f)
<div>
    <label for="id_name">Name:</label>
<div class="helptext" id="id_name_helptext">Help Text!</div>
<input type="text" name="name" value="David" required aria-describedby="id_name_helptext" id="id_name">
</div>

In [9]: f.add_error("name", "some additional error")

In [10]: print(f)
<div>
    <label for="id_name">Name:</label>
<div class="helptext" id="id_name_helptext">Help Text!</div>
<ul class="errorlist" id="id_name_error"><li>some additional error</li></ul>
<input type="text" name="name" value="David" required aria-invalid="true" aria-describedby="id_name_helptext" id="id_name">
</div>

```

Reply:
Ok, let's have a test that covers this case
I feel like it could be cached in the help text only case and then, we shouldn't cache it for errors which are more dynamic